tests/async/test_dispatch_event.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) Microsoft Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+async def test_should_dispatch_click_event(page, server):
+    await page.goto(server.PREFIX + '/input/button.html')
+    await page.dispatchEvent('button', 'click')
```

Comment:
This currently fails 💥 due to (I think) a serialization error: `eventInit: expected object, got null`. 

Full trace:

```
――――――――――――――――――――――――――――――――――――――――――――――――――――――――――― test_should_dispatch_click_event[chromium] ―――――――――――――――――――――――――――――――――――――――――――――――――――――――――――

page = <playwright.async_api.Page object at 0x1040f14e0>, server = http://localhost:60996

    async def test_should_dispatch_click_event(page, server):
        await page.goto(server.PREFIX + '/input/button.html')
>       await page.dispatchEvent('button', 'click')

tests/async/test_dispatch_event.py:18: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
playwright/async_api.py:3196: in dispatchEvent
    selector=selector, type=type, eventInit=eventInit, timeout=timeout
playwright/page.py:300: in dispatchEvent
    return await self._main_frame.dispatchEvent(**locals_to_params(locals()))
playwright/frame.py:241: in dispatchEvent
    "dispatchEvent", dict(selector=selector, type=type, eventInit=eventInit)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <playwright.connection.Channel object at 0x1040f1710>, method = 'dispatchEvent', params = {'eventInit': None, 'selector': 'button', 'type': 'click'}

    async def send(self, method: str, params: dict = None) -> Any:
        if params is None:
            params = {}
        callback = self._connection._send_message_to_server(self._guid, method, params)
>       result = await callback.future
E       playwright.helper.Error: ('eventInit: expected object, got null', 'Error: eventInit: expected object, got null\n    at /snapshot/driver/node_modules/playwright/lib/rpc/validatorPrimitives.js:73:19\n    at /snapshot/driver/node_modules/playwright/lib/rpc/validator.js:30:20\n    at Object.FrameDispatchEventParams (/snapshot/driver/node_modules/playwright/lib/rpc/validatorPrimitives.js:78:27)\n    at DispatcherConnection._validateParams (/snapshot/driver/node_modules/playwright/lib/rpc/server/dispatcher.js:114:32)\n    at DispatcherConnection.dispatch (/snapshot/driver/node_modules/playwright/lib/rpc/server/dispatcher.js:136:36)\n    at Transport.onmessage (/snapshot/driver/node_modules/playwright/lib/rpc/server.js:35:55)\n    at Immediate._onImmediate (/snapshot/driver/node_modules/playwright/lib/rpc/transport.js:64:26)\n    at processImmediate (internal/timers.js:456:21)')

playwright/connection.py:39: Error
```

Reply:
here's the definition from protocol:

```
    dispatchEvent:
      parameters:
        type: string
        eventInit: SerializedArgument
```

Reply:
@pavelfeldman Should the type of `eventInit` be changed to optional (`SerializedArgument?`) or should we not be getting null here?

Reply:
`dispatchEvent` should serialize argument to conform to the protocol:

```py
async def dispatchEvent(self, type: str, eventInit: Dict = None) -> None:
        await self._channel.send("dispatchEvent", dict(type=type, eventInit=serialize_argument(eventInit)))
```

Reply:
Thank you! I completed the impl in frame and element handle.